galasa-parent/dev.galasa.framework.api.runs/src/main/java/dev/galasa/framework/api/runs/internal/ResultArchive.java

```diff
@@ -0,0 +1,115 @@
+package dev.galasa.framework.api.runs.internal;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.function.Consumer;
+import java.util.zip.GZIPInputStream;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.osgi.service.component.annotations.ServiceScope;
+
+import dev.galasa.framework.spi.IFramework;
+import dev.galasa.framework.spi.IResultArchiveStoreDirectoryService;
+import dev.galasa.framework.spi.IRunResult;
+import dev.galasa.framework.spi.utils.GalasaGsonBuilder;
+
+@Component(service = Servlet.class, scope = ServiceScope.PROTOTYPE, property = {
+        "osgi.http.whiteboard.servlet.pattern=/resultarchive" }, name = "Galasa RAS")
+public class ResultArchive extends HttpServlet {
+    private static final long serialVersionUID = 1L;
+
+    private Log logger = LogFactory.getLog(getClass());
+
+    private final Gson gson = GalasaGsonBuilder.build();
+
+    @Reference
+    public IFramework framework; // NOSONAR
+
+    @Override
+    public void doPost(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+        JsonObject requestBody = gson.fromJson(new InputStreamReader(req.getInputStream()), JsonObject.class);
```

Comment:
you should be doing this as RESTful calls,   not passing everything via json objects.